mgo.go

```diff
@@ -173,10 +174,38 @@ func generatePEM(path string, serverCert *x509.Certificate, serverKey *rsa.Priva
 
 // Start starts a MongoDB server in a temporary directory.
 func (inst *MgoInstance) Start(certs *Certs) error {
-	dbdir, err := ioutil.TempDir("", "test-mgo")
+	var err error
+	mongopath, _, err := installedMongod.Get()
 	if err != nil {
 		return err
 	}
+
+	dbdir := ""
+
+	// Check for snap confined mongod.
+	if mongopath == "/snap/bin/juju-db.mongod" {
```

Comment:
Is it possible to have both the snap and the apt version installed and attempting to use the apt version for tests (e.g. testing the 2.7 branch)? This change will always give precedence to the snap mongo. If that's the case perhaps we would need to have a version preference argument or check for a magic envvar

Reply:
JUJU_MONGOD env var has preference over everything, so that could be the solution.

Reply:
Also I don't think we'll be updating 2.7 anytime soon, or even updating the 2.7 import for this package. So safe to say we can go ahead with this.